### ISSUE TITLE
matches-prop-shorthand should not complain when matching two internal properties

### ISSUE THREAD

**Issue by demands** [wix/eslint-plugin-lodash#127]
I've got code like this:

```js
_.reject(list, elem => elem.propA === elem.propB);
```

It fails the `lodash/matches-prop-shorthand` check, despite the fact that (as far as I know!) it's impossible to use matchesProp shorthand in this case.

**Comment by captbaritone** [comment]
Good catch. This looks like a bug to me.

**Comment by ganimomer** [comment]
That is a bug.
Fixed in `v2.2.3`.
Thanks for reporting this!
